openedx-unsupported/configuration
# Define role to deploy blockstore

This PR adds a new role to deploy blockstore service. Deploying this service would be optional and would depend on the variable `BLOCKSTORE_ENABLED` to be set to `true`. By default, the value is set to `false`.

**Testing Instructions:**

1. Since the service is not configured with the OCIM, the required database and users need to be created. Once the database and users are created, the variables `BLOCKSTORE_DEFAULT_DB_NAME`, `BLOCKSTORE_DATABASE_HOST`, `BLOCKSTORE_DATABASE_USER`, `BLOCKSTORE_DATABASE_PASSWORD` will need to be overridden with the appropriate values.
2. Checkout the changes from this branch and deploy a sandbox by passing the following variables:
```
BLOCKSTORE_ENABLED: true
BLOCKSTORE_DATABASE_HOST: <mysql-server-to-be-used>
BLOCKSTORE_DATABASE_USER: <created-username>
BLOCKSTORE_DATABASE_PASSWORD: <created-password>
BLOCKSTORE_DEFAULT_DB_NAME: <created-database>
```
3. Once the sandbox is deployed successfully, verify that the service is running in your appserver and test the apis.

**Reviewers:**


---

Make sure that the following steps are done before merging:
